iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -176,6 +176,18 @@ void CachedConnection::SyncAttachDbs() {
 // @bsimethod
 //---------------------------------------------------------------------------------------
 void CachedConnection::Execute(std::function<void(QueryAdaptorCache&,RunnableRequestBase&)> cb, std::unique_ptr<RunnableRequestBase> request) {
+    if (m_db.IsDbOpen()) {
+        // Check if the primary file data version has changed
+        uint32_t dataVersion;
+        if (BE_SQLITE_OK == GetPrimaryDb().GetFileDataVersion(dataVersion)) {
+            if (dataVersion != m_primaryFileDataVer) {
+                m_db.ClearECDbCache();
+                m_db.ClearDbCache();
+                m_primaryFileDataVer = dataVersion;
+            }
+        }
+    }

```

Comment:
Is this something we should be doing in ECPresentation as well with our secondary connections?

Reply:
Sounds good. I don't think we're going to make a switch to instance queries in the native ECPresentation library, but I'll keep this in mind if we do.